addons/crm/tests/test_crm_lead.py

```diff
@@ -119,6 +119,31 @@ def test_crm_lead_creation_partner(self):
         self.assertEqual(lead.zip, self.contact_1.zip)
         self.assertEqual(lead.country_id, self.contact_1.country_id)
 
+    @users('user_sales_manager')
+    def test_crm_lead_suggested_recipient_partner_no_email(self):
+      partner = self.contact_2
+      lead = self.env['crm.lead'].create({
+        'name': 'Test Lead',
+        'partner_id': partner.id,
+        'email_cc': 'test_cc@odoo.com'
+      })
+      result_with_email = lead._message_get_suggested_recipients()
+      expected_result_with_email = [
+          (False, 'test_cc@odoo.com', 'CC Email'),
+          (partner.id, partner.email, 'Customer')
+        ]
+      self.assertEqual(result_with_email[lead.id], expected_result_with_email)
+
+      partner.write({
+        'email': False
+      })
+      result_no_email = lead._message_get_suggested_recipients()
+      expected_result_no_email = [
+        (False, 'test_cc@odoo.com', 'CC Email'),
+        (partner.id, partner.name, 'Customer')
+      ]
```

Comment:
See the issue here is that it's still finding the partner of `email_from`

That's because the compute for email_from is stored, and not updated if (notably) the email is entirely removed.
`if lead.partner_id.email and lead._get_partner_email_update():`

Would need to start by making the email False on the partner, then assign them to the lead, then assign an email to the partner. Then the test should work fine.